exercises/practice/leap/.approaches/introduction.md

```diff
@@ -0,0 +1,96 @@
+# Introduction
+
+Every fourth year is a leap year (with some exceptions), but let's consider this one condition first. 
+
+To solve the Leap problem, we must determine if a year is evenly divisible by a number or if a reminder of an integer division is zero. 
+Such operation in computing is called [modulo][modulo]. 
+
+Unlike many languages, Elixir does not have [operators][operators] for either integer division or modulo. 
+Instead, it provides [`rem/2`][rem] guard and the [`Integer.mod/2`][mod] function.
+
+The two functions differ in how they work with negative numbers, but since, in this exercise,
+all the numbers are non-negative, both could work, depending on the approach.
+
+## General Solution
+
+To check if a year is divisible by `n`, we can do `rem(year, n) == 0`. We can define a function to make the intent clearer. 
+
+```elixir
+defp divides?(number, divisor), do: rem(number, divisor) == 0 
+```
+
+Any approach to the problem will perform this check three times to check if a year is equally divisible by 4, 100 and 400.
+What will differ is what Elixir features we will use to combine the checks. 
+
+## Approach: Boolean Operators
+
+The full rules are as follows:
+A year is a leap year if 
+* it is divisible by 4 
+* but not divisible by 100
+* unless it is divisible by 400
+
+We can use [boolean operators][boolean-operators] to combine the checks, for example, like so: 
+
+```elixir
+divides?(year, 400) or (not(divides?(year, 100))) and divides?(year, 4)
+```
+In the [boolean operators appraoch][operators-approach] we discuss the details of the solution.
+It includes variations of the operators and their precendence.
+
+## Approach: Function Guards
+
+Instead of using boolean operators, we can define multiple `leap_year?/1` functions with different guards. 
+We can use the order of the definitions to ensure correct check.
+
+```elixir
+def leap_year?(year) when rem(year, 400) == 0, do: true
+def leap_year?(year) when rem(year, 100) == 0, do: false
+def leap_year?(year) when rem(year, 4) == 0, do: true
+def leap_year?(_), do: false
+```
+
+In the [functions with guards approach][guards-approach] we discuss why in this approach the `Integer.mod/2` function will not work. 
+
+## Approach: Using cond
+
+Similarly to the functions with guards, the order of the checks can be done inside a function with `cond` expression. 
+
+```elixir
+cond do
+    divides?(year, 400) -> true
+    divides?(year, 100) -> false
+    divides?(year, 4) -> true
+    true -> false
+end
+```
+
+We discuss this briefly in the [cond approach][cond-approach]
+
+## Approach: Using case
```

Comment:
This approach should be called
```suggestion
## Approach: Using case on a tuple
```

There are different solutions that could be called "using case", for example any `cond` can be rewritten to a `case` (but this one is not very good, so let's not describe it 😉)

```elixir
case year do
  year when rem(year, 400) == 0 -> true
  year when rem(year, 100) == 0 -> false
  year when rem(year, 4) == 0 -> true
  _ -> false
end
```

Reply:
I plan to show multiple options, similar to what I've done in the boolean operators approach. the `on a tuple` will be just one variant.

Reply:
Hmm... I'm not entirely sure what the philosophy of writing Exercism approaches is supposed to be because I don't really read them. Are they supposed to cover every possible way of solving an exercise, or just the most common or interesting ones?

Because if we only want the common ones, then "case on a tuple" is the only case-based approach that I would suggest showing, based on what the community solutions tend to be. The rest of the case approaches that I see in the current version of the `exercises/practice/leap/.approaches/flow/content.md` file are less appropriate uses of `case` that should have been an `if` or a `cond` instead (like you point out already).